File: bdp-master-client/src/main/scala/com/github/bdp/master/client/service/AlertIndexService.scala
package com.github.bdp.master.client.service

import com.github.bdp.master.client.Constants._
import com.github.bdp.master.client.domain.AlertIndex
import com.github.bdp.master.client.util.{JsonDecoder, RedisClient}
import com.typesafe.scalalogging.LazyLogging

object AlertIndexService extends LazyLogging {

  def getAlertIndexBy(id: Long): AlertIndex = {
    JsonDecoder.decodeAlertIndex(RedisClient.get(s"$ALERT_INDEX_KEYSPACE:$id"))
  }

  def getAlertIndexBy(name: String): AlertIndex = {
    val key = RedisClient.get(s"i_$ALERT_INDEX_KEYSPACE:$name")
		JsonDecoder.decodeAlertIndex(RedisClient.get(key))
  }
}

File: bdp-master-server/src/main/java/com/github/bdp/master/server/service/AlertIndexService.java
package com.github.bdp.master.server.service;


import com.github.bdp.master.server.domain.AlertIndex;

import java.util.List;

public interface AlertIndexService {

	AlertIndex findOne(Long id);

	AlertIndex findOne(String alertName);

	void save(AlertIndex alertIndex);

	void delete(Long id);

	List<AlertIndex> findAll();

	void loadAll();
}
